controllers/front/Result.php

```diff
@@ -27,33 +27,50 @@
 // phpcs:disable PSR1.Classes.ClassDeclaration
 
 use Adyen\PrestaShop\service\adapter\classes\ServiceLocator;
+use Adyen\PrestaShop\controllers\FrontController;
+use Adyen\AdyenException;
 
-class AdyenResultModuleFrontController extends \Adyen\PrestaShop\controllers\FrontController
+class AdyenResultModuleFrontController extends FrontController
 {
     /**
      * @var bool
      */
     public $ssl = true;
 
     /**
-     * @throws \Adyen\AdyenException
+     * @throws Adapter_Exception
+     * @throws AdyenException
      */
     public function postProcess()
     {
-        if (!isset($_SESSION['paymentData'])) {
+        // Retrieve cart based on the reference parameter
+        $cart = new \Cart(\Tools::getValue('reference'));
```

Comment:
This `reference` seems too generic and I fear it might be overridden

Reply:
maybe merchantReference? 
I was thinking about the same with the reference but I also didn't want to include our naming

Reply:
`merchantReference` seems a bit better, although I was thinking about prefixing it with `adyen`. So `adyenMerchantReference`?

Reply:
I like the adyenMerchantReference - going to switch to that